ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -5522,6 +5525,9 @@ NtFreeVirtualMemory(IN HANDLE ProcessHandle,
                         goto FailPath;
                     }
 
+                    // Charge quota for the new VAD
+                    PsChargeProcessNonPagedPoolQuota(Process, sizeof(MMVAD_LONG));
```

Comment:
In the future as I will be going to improve the quota support in the kernel with a second overhaul of the code, the ReactOS kernel will complain if the quota limits have been exceeded. 

With that being said, it's good practice that you check for `STATUS_QUOTA_EXCEEDED` and handle the failure case appropriately. My review comment applies the same for all your other code paths where `PsChargeProcessNonPagedPoolQuota` is used.